test/runtests.jl

```diff
@@ -426,3 +426,17 @@ quadgk_segbuf_printnull(args...; kws...) = quadgk_segbuf_print(devnull, args...;
     @inferred QuadGK.to_segbuf([0,1])
     @inferred QuadGK.to_segbuf([(0,1+3im)])
 end
+
+# Extension package only supported in 1.9+
+@static if VERSION >= v"1.9"
+    using Enzyme
+    f1(x) = quadgk(cos, 0., x)[1]
+    f2(x) = quadgk(cos, x, 1)[1]
+    f3(x) = quadgk(y->cos(x * y), 0., 1.)[1]
+
+    @testset "Enzyme" begin
+        @test cos(0.3) ≈ Enzyme.autodiff(Reverse, f1, Active(0.3))[1][1]
+        @test -cos(0.3) ≈ Enzyme.autodiff(Reverse, f2, Active(0.3))[1][1]
+        @test (0.3 * cos(0.3) - sin(0.3))/(0.3*0.3) ≈ Enzyme.autodiff(Reverse, f3, Active(0.3))[1][1]
```

Comment:
Would be good to check that it works for something like `quadgk_count` that wraps `quadgk`, though I expect that Enzyme will have no trouble with this.

I'm guessing that in-place integrands (ala `quadgk!`) and batch integrands will still not work.  Maybe add a `@test_broken` for these as a reminder.

Reply:
Just to make sure I call those properly, do you have a sample usage function of these?

Reply:
Your f1–3 should still work with `quadgk_count` — it just returns a tuple of three items `I, E, count` rather than a tuple of two items `I, E` like `quadgk`

Reply:
Done, and all pass. While at it, I also fixed the count version to be more performant/compatible. Modifying count in place in the closure results in a Core.Box type, which is type unstable, but now is a stable ref value.